src/Protocol/ActivityPub/Processor.php

```diff
@@ -389,6 +389,26 @@ private static function processContent($activity, $item)
 		return $item;
 	}
 
+	/**
+	 * Generate a GUID out of an URL
+	 *
+	 * @param string $url message URL
+	 * @return string with GUID
+	 */
+	private static function getGUIDByURL(string $url)
+	{
+		$parsed = parse_url($url);
+
+		$host_hash = hash('crc32', $parsed['host']);
+
+		unset($parsed["scheme"]);
+		unset($parsed["host"]);
+
+		$path = implode("/", $parsed);
+
+		return $host_hash . '-'. hash('fnv164', $path) . '-'. hash('joaat', $path);
```

Comment:
Where did you pull these algorithms from?

Reply:
I checked which hash methods are available for PHP and used two that had a nice length. I used that code that I took from the PHP help page:

```
$data = "hello";
foreach (hash_algos() as $v) {
        $r = hash($v, $data, false);
        printf("%-12s %3d %s\n", $v, strlen($r), $r);
}

```

Reply:
Awesome!